src/task.rs

```diff
@@ -0,0 +1,149 @@
+use std::future::Future;
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::task::Poll;
+use tokio::task;
+
+use tracing::{debug, error};
+
+use enostr::RelayPool;
+use nostrdb::{Filter, Ndb, Transaction};
+use uuid::Uuid;
+
+use crate::note::NoteRef;
+use crate::Damus;
+
+pub fn spawn_track_user_relays(damus: &mut Damus) {
+    // This is only safe because we are absolutely single threaded ...
+    let damus_ptr = &mut *damus as *mut Damus;
+    spawn_sendable(async move {
```

Comment:
I think this implementation for adding relays can be simplified into a synchronous implementation that runs every frame in `app::update_damus`.

In the case where the user is actively using the app, the time between frames is going to be so low that it won't matter that we'll have to wait until
the next frame to check ndb for a new relay list. In the case where the user isn't using the app, the update function still runs, about once per second
it seems so if the user is just staring at their relay list without any interaction and an update to their list occurs, they will see it, albeit with
about a second delay.

Do you see a reason that I don't for the increased complexity of this async impl?

Reply:
I think polling for activity doesn't scale well to apps with lots of things to poll for.  So it might be fine to poll for user relay lists, but when we add many many other things it will be less great